desktop-widgets/printer.cpp

```diff
@@ -150,8 +150,12 @@ void Printer::render(int pages)
 
 	QSize originalSize = profile->size();
 	if (collection.count() > 0) {
-		printFontScale = (double)collection.at(0).geometry().size().height() / (double)profile->size().height();
-		profile->resize(collection.at(0).geometry().size());
+		// A "standard" profile has about 600 pixels in height.
+		// Scale the fonts in the printed profile accordingly.
+		// This is arbitrary, but it seems to work reasonably.
+		QSize size = collection[0].geometry().size();
+		printFontScale = size.height() / 600.0;
+		profile->resize(size);
```

Comment:
I can see the logic - but I think the whole size logic of 'everything' for printing is just a hack on top of a kludge on top of a bad design...
Happy to merge this as it doesn't make things worse AFAICT, but the icon sizes and font sizes really need to be fixed - because fonts aren't the only things that are using weird heuristics - we have so many reports that the tank change icons are way off at times.